### PR TITLE
docstring fixes to pybamm.plotting.quick_plot

### PR DESCRIPTION
**Pull request by neilsengupta-elysia**
# Description

Practise contributing to issue for Hackathon:

In the file pybamm.plotting.quick_plot.py, added "optional" keyword to "step" argument in dynamic_plot function. Removed round brackets around optional keywords in the arguments "number_of_images", "duration", "output_filename" in create_gif function.

Related #3392 - Improve documentation

## Type of change
Documentation Update

### CODE REVIEW

File: pybamm/plotting/quick_plot.py [hunk 1]
```diff
@@ -464,6 +464,8 @@ def plot(self, t, dynamic=False):
         ----------
         t : float
             Dimensional time (in 'time_units') at which to plot.
+        dynamic : bool
+
```
Comment (brosaplanella) [inline pybamm/plotting/quick_plot.py:468]: You could include a short description, e.g. "Whether the plot should be dynamic or not"

**Review by brosaplanella — COMMENTED** [review]
Just a small comment, but looks good overall!

File: pybamm/plotting/quick_plot.py (hunk 1 above)
Comment (kratman) [reply pybamm/plotting/quick_plot.py:468]: It is also optional

**Review by kratman — COMMENTED** [review]
While the syntax used in your issue description
```
Fixes # Issue 3392 - Improve documentation (partial fix)
```
will not work because it is incorrectly formatted, you should not mark a partial fix as with the `fixes` tag. When used correctly the `fixes` keyword causes the issue to be closed. This would me that this 10 line change would close a much larger ticket.

Usually you can do
```
Related #3392
```
to tag the issue without closing it

File: pybamm/plotting/quick_plot.py (hunk 1 above)
Comment (neilsengupta-elysia) [reply pybamm/plotting/quick_plot.py:469]: added a short description, as well as the optional tag.

**Comment by neilsengupta-elysia** [comment]
Fixed PR description to remove "fixes" keyword

**Review by agriyakhetarpal — APPROVED** [review]

**Comment by kratman** [comment]
@neilsengupta-elysia I updated your PR description. Having `Related # issue 3392` does not work, it needs to be `Related #3392`. If there are any spaces between # and the number it will not link properly

**Review by kratman — APPROVED** [review]

**Pull request merged**